port/main.c

```diff
@@ -577,12 +95,6 @@ int main(int argc, char **argv)
 #endif
 #endif
 
-	mutexCreate(&rt_table.lock);
-
-#if LWIP_EXT_PF
-	init_filters();
```

Comment:
I'm not sure if PF shouldn't be initialised before netif is created – it's create mutex. Not sure what trigger pf to work. 
@nalajcie do you remember any details?

Reply:
Thanks for pointing this out. Technically PF mutex should be initialized before driver thread is started (PF is called on received packets). It was broken in old code as well.